101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1118 @@
+            }
+        },
+        "timeZoneID": {
+            "type": "string",
+            "defaultValue": "UTC",
+            "metadata": {
+                "description": "TimeZone ID to be used on VMs. Get available timezones with powershell Get-TimeZone command."
+            }
+        },
+        "externalDnsZone": {
+            "type": "string",
+            "defaultValue": "contosocorp.com",
+            "metadata": {
+                "description": "External public DNS domain zone. NOT AD domain. This the external domain your certs will point to."
+            }
+        },
+        "deployHA": {
+            "type": "bool",
+            "defaultValue": false,
+            "metadata": {
+                "description": "This will trigger certificate request and HA deployment. If set to false, will not create HA deployment nor request certificates."
+            }
+        },
+        "dcCount": {
+            "type": "int",
+            "defaultValue": 1,
+            "metadata": {
+                "description": "How many Domain Controllers would you like to deploy?"
+            }            
+        },        
+        "rdcbCount": {
+            "type": "int",
+            "defaultValue": 1,
+            "metadata": {
+                "description": "How many RD Connection Brokers would you like to deploy?"
+            }
+        },
+        "rdwgCount": {
+            "type": "int",
+            "defaultValue": 1,
+            "metadata": {
+                "description": "How many RD Web Access/Gateways would you like to deploy?"
+            }            
+        },
+        "rdshCount": {
+            "type": "int",
+            "defaultValue": 1,
+            "metadata": {
+                "description": "How many RD Session Hosts would you like to deploy?"
+            }            
+        },
+        "lsfsCount": {
+            "type": "int",
+            "defaultValue": 1,
+            "metadata": {
+                "description": "How many License/File Servers would you like to deploy?"
+            }            
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A2_v2",
+            "allowedValues": [
+                "Standard_A2_v2",
+                "Standard_A4_v2",
+                "Standard_A8_v2",
+                "Standard_D1_v2",
+                "Standard_D2_v2",
+                "Standard_D3_v2",                
+                "Standard_D2_v3",
+                "Standard_D4_v3"
+            ],
+            "metadata": {
+                "description": "What is the VM size for all VMs?"
+            }            
+        },
+        "vmStorageSkuType": {
+            "type": "string",
+            "defaultValue": "Standard_LRS",
+            "allowedValues": [
+                "StandardSSD_LRS",
+                "Standard_LRS",
+                "Premium_LRS"

```

Comment:
the allowedValues for VM size prevent you from using premium disk...  If you want to use premium, you need to restrict sizes to those that allow for it...